sentry/src/main/java/io/sentry/clientreport/AtomicClientReportStorage.java

```diff
@@ -0,0 +1,80 @@
+package io.sentry.clientreport;
+
+import io.sentry.SentryLevel;
+import io.sentry.SentryOptions;
+import io.sentry.transport.DataCategory;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+final class AtomicClientReportStorage implements ClientReportStorage {
```

Comment:
`m`: I know it's a draft PR, but I think we are missing some tests for this class.

Reply:
It's only tested indirectly via `ClientReportTest`. Gonna add a test for it.